### DISCUSSION TITLE
signTypedData (eth_signTypedData_v4) not available

### DISCUSSION THREAD

**Discussion by bartomolina** [WalletConnect/walletconnect-monorepo#3725] [Developer Support] [answered]
I'm building an app that uses wagmi's signTypedData. It works fine on the desktop when it's connected to Metamask, but I couldn't make it work through Wallet Connect (on mobile or a browser without metamask installed). This is the error I'm receiving:

```
MethodNotFoundRpcError: The method does not exist / is not available.
Details: the method eth_signTypedData_v4 does not exist/is not available
Version: viem@1.12.2
```

I've tried with different wallets (Trust, Metamask, AlphaWallet), but no luck. Any ideas on why it may be failing?

<img width="1025" alt="Screenshot 2023-09-29 at 18 50 25" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/827672/8adb55d3-fb0a-46ee-b4a1-eca4174d1bbd">

**Comment by glitch-txs** [comment, marked as answer]
Hi for now use two chains minimum on your chains array for the WalletConnect connector.
If you're using mainnet this should look like
```ts
const chains = [mainnet, mainnet]
```

We're currently working on a fix for this. Thanks!

**Comment by xulian0x** [comment]
was this solved, getting a similar issue on this where users suddenly are having issues with connecting via QR code.
